foam_controlwrapper/foam_mesh.py

```diff
@@ -0,0 +1,655 @@
+            # init objectRegistry and map to mesh name
+            foamface.init(name, os.path.abspath(os.path.join(self.path,
+                                                             os.pardir)))
+            # add mesh to objectRegisty
+            foamface.addMesh(name, pointCoordinates, cellPoints, facePoints,
+                             patchNames, patchFaces)
+            # write mesh to disk
+            foamface.writeMesh(name)
+
+            # write possible cell data to time directory
+            for cell in mesh.iter_cells():
+                self.update_cell(cell)
+
+    def get_point(self, uuid):
+        """Returns a point with a given uuid.
+
+        Returns the point stored in the mesh
+        identified by uuid. If such point do not
+        exists an exception is raised.
+
+        Parameters
+        ----------
+        uuid
+            uuid of the desired point.
+
+        Returns
+        -------
+        Point
+            Mesh point identified by uuid
+
+        Raises
+        ------
+        Exception
+            If the point identified by uuid was not found
+
+        """
+
+        try:
+            coords = foamface.getPointCoordinates(self.name,
+                                                  self._uuidToFoamLabel[uuid])
+            point = Point(coords, uuid)
+            return point
+        except KeyError:
+            error_str = "Trying to get an non-existing point with uuid: {}"
+            raise ValueError(error_str.format(uuid))
+
+    def get_edge(self, uuid):
+        """Returns an edge with a given uuid.
+
+        Returns the edge stored in the mesh
+        identified by uuid. If such edge do not
+        exists an exception is raised.
+
+        Parameters
+        ----------
+        uuid
+            uuid of the desired edge.
+
+        Returns
+        -------
+        Edge
+            Edge identified by uuid
+
+        Raises
+        ------
+        Exception
+            If the edge identified by uuid was not found
+
+        """
+        message = "Edges are not supported yet in OpenFoam engine"
+        raise NotImplementedError(message)
+
+    def get_face(self, uuid):
+        """Returns a face with a given uuid.
+
+        Returns the face stored in the mesh
+        identified by uuid. If such face do not
+        exists an exception is raised.
+
+        Parameters
+        ----------
+        uuid
+            uuid of the desired face.
+
+        Returns
+        -------
+        Face
+            Face identified by uuid
+
+        Raises
+        ------
+        Exception
+            If the face identified by uuid was not found
+
+        """
+
+        try:
+            pointLabels = foamface.getFacePoints(self.name,
+                                                 self._uuidToFoamLabel[uuid])
+            puids = [self._foamPointLabelToUuid[lbl] for lbl in pointLabels]
+            face = Face(puids, uuid)
```

Comment:
the faces provided by `iter_face` seem to have `data` with a `LABEL`.   `get_face` don't have a label.    the faces provided by these two methods (`iter_face` and `get_face`) should be the same.